core/src/main/scala/org/apache/spark/sql/execution/ObjectHashMapAccessor.scala

```diff
@@ -304,32 +305,42 @@ case class ObjectHashMapAccessor(@transient session: SnappySession,
     val updateMinMax = integralKeys.map { index =>
       s"$hashMapTerm.updateLimits(${keyVars(index).value}, $index);"
     }.mkString("\n")
+    // for SnappyData column or row tables there is no need to make
+    // a copy of non-primitive values into the map
+    var hasExchange = false
+    val doCopy = child.find {
```

Comment:
Your comment says that for column table and row tables doCopy should be false but it returns true. Also, I must be missing something but child.find finds any node that matches one of these three conditions. So, it will always match the last condition.